{"cells":[{"cell_type":"code","metadata":{"tags":[],"cell_id":"00000-56e200f1-ceab-44ba-a1a1-5de8b455da54","deepnote_to_be_reexecuted":false,"source_hash":"815a9989","execution_millis":1152,"execution_start":1614115830003,"deepnote_cell_type":"code"},"source":"# imports\nimport os\n\nimport pandas as pd\nimport matplotlib.pyplot as plt\nimport numpy as np\n\n#import specific sklearn packages\nfrom sklearn.model_selection import train_test_split\n","execution_count":1,"outputs":[]},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00002-cebcc986-6637-446e-8698-1eee14e7cd96","deepnote_to_be_reexecuted":false,"source_hash":"c2c03afe","execution_millis":3937,"execution_start":1614115831159,"deepnote_cell_type":"code"},"source":"# install kaggle to use Kaggle API\n!pip install kaggle","execution_count":2,"outputs":[{"name":"stdout","text":"Collecting kaggle\n  Downloading kaggle-1.5.10.tar.gz (59 kB)\n\u001b[K     |████████████████████████████████| 59 kB 7.3 MB/s \n\u001b[?25hRequirement already satisfied: six>=1.10 in /shared-libs/python3.7/py-core/lib/python3.7/site-packages (from kaggle) (1.15.0)\nRequirement already satisfied: certifi in /shared-libs/python3.7/py/lib/python3.7/site-packages (from kaggle) (2020.12.5)\nRequirement already satisfied: python-dateutil in /shared-libs/python3.7/py-core/lib/python3.7/site-packages (from kaggle) (2.8.1)\nRequirement already satisfied: requests in /shared-libs/python3.7/py/lib/python3.7/site-packages (from kaggle) (2.25.1)\nRequirement already satisfied: tqdm in /shared-libs/python3.7/py/lib/python3.7/site-packages (from kaggle) (4.57.0)\nRequirement already satisfied: python-slugify in /shared-libs/python3.7/py/lib/python3.7/site-packages (from kaggle) (4.0.1)\nRequirement already satisfied: urllib3 in /shared-libs/python3.7/py/lib/python3.7/site-packages (from kaggle) (1.26.3)\nRequirement already satisfied: chardet<5,>=3.0.2 in /shared-libs/python3.7/py/lib/python3.7/site-packages (from requests->kaggle) (3.0.4)\nRequirement already satisfied: idna<3,>=2.5 in /shared-libs/python3.7/py/lib/python3.7/site-packages (from requests->kaggle) (2.10)\nRequirement already satisfied: text-unidecode>=1.3 in /shared-libs/python3.7/py/lib/python3.7/site-packages (from python-slugify->kaggle) (1.3)\nBuilding wheels for collected packages: kaggle\n  Building wheel for kaggle (setup.py) ... \u001b[?25ldone\n\u001b[?25h  Created wheel for kaggle: filename=kaggle-1.5.10-py3-none-any.whl size=73268 sha256=13daedc72306704f532bb3ceb68c2c42f100e3133a10cada05d37e34509b4f60\n  Stored in directory: /root/.cache/pip/wheels/ea/c5/fe/7e7fb5b3d1f150fac96188949b3d83d375a4c9df16ba557e52\nSuccessfully built kaggle\nInstalling collected packages: kaggle\nSuccessfully installed kaggle-1.5.10\n\u001b[33mWARNING: You are using pip version 20.1.1; however, version 21.0.1 is available.\nYou should consider upgrading via the '/root/venv/bin/python -m pip install --upgrade pip' command.\u001b[0m\n","output_type":"stream"}]},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00002-77feab85-c920-4795-87b7-12aae9c910b1","deepnote_to_be_reexecuted":false,"source_hash":"9d5eae84","execution_millis":1329,"execution_start":1614115835106,"deepnote_cell_type":"code"},"source":"!mkdir .kaggle\n!chmod 600 ~/.kaggle/kaggle.json","execution_count":3,"outputs":[{"name":"stdout","text":"mkdir: cannot create directory ‘.kaggle’: File exists\nchmod: cannot access '/root/.kaggle/kaggle.json': No such file or directory\n","output_type":"stream"}]},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00003-35cfeb8d-4a15-4713-9d49-b20e4893ec35","deepnote_to_be_reexecuted":false,"source_hash":"1bda9d17","execution_millis":964,"execution_start":1614115836444,"deepnote_cell_type":"code"},"source":"!kaggle datasets list -s competitive-data-science-predict-future-sales","execution_count":4,"outputs":[{"name":"stdout","text":"Traceback (most recent call last):\r\n  File \"/root/venv/bin/kaggle\", line 5, in <module>\r\n    from kaggle.cli import main\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/__init__.py\", line 23, in <module>\r\n    api.authenticate()\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/api/kaggle_api_extended.py\", line 166, in authenticate\r\n    self.config_file, self.config_dir))\r\nOSError: Could not find kaggle.json. Make sure it's located in /root/.kaggle. Or use the environment method.\r\n","output_type":"stream"}]},{"cell_type":"markdown","source":"For whatever reason **Kaggle doesn't have the reference link for the datasets** listed under this competition. But as you can see from above, *some Kaggle users have translated* the mostly Russian fields so they're understandable. But they only did it for 3/6 of the files... \n- ~~remisharoon/predict-future-sales-translated-dataset~~  \n- ywhenlyn/predict-future-sales-translated-to-english ","metadata":{"tags":[],"cell_id":"00004-560b2c63-94d4-4df2-92a6-c71bff2318f2","deepnote_cell_type":"markdown"}},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00004-40a37d77-10de-46d2-b791-943d66529947","deepnote_to_be_reexecuted":false,"source_hash":"9a4926da","execution_millis":960,"execution_start":1614115837407,"deepnote_cell_type":"code"},"source":"!kaggle competitions files competitive-data-science-predict-future-sales","execution_count":5,"outputs":[{"name":"stdout","text":"Traceback (most recent call last):\r\n  File \"/root/venv/bin/kaggle\", line 5, in <module>\r\n    from kaggle.cli import main\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/__init__.py\", line 23, in <module>\r\n    api.authenticate()\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/api/kaggle_api_extended.py\", line 166, in authenticate\r\n    self.config_file, self.config_dir))\r\nOSError: Could not find kaggle.json. Make sure it's located in /root/.kaggle. Or use the environment method.\r\n","output_type":"stream"}]},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00005-203414a7-90d0-4496-84b4-f1ba25de6444","deepnote_to_be_reexecuted":false,"source_hash":"3e1a85e4","execution_millis":980,"execution_start":1614115838376,"deepnote_cell_type":"code"},"source":"!kaggle competitions download -w competitive-data-science-predict-future-sales -p /work/predict_future_sales/files","execution_count":6,"outputs":[{"name":"stdout","text":"Traceback (most recent call last):\r\n  File \"/root/venv/bin/kaggle\", line 5, in <module>\r\n    from kaggle.cli import main\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/__init__.py\", line 23, in <module>\r\n    api.authenticate()\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/api/kaggle_api_extended.py\", line 166, in authenticate\r\n    self.config_file, self.config_dir))\r\nOSError: Could not find kaggle.json. Make sure it's located in /root/.kaggle. Or use the environment method.\r\n","output_type":"stream"}]},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00002-dfa13975-f461-4b15-afd5-a0dd014fcbc6","deepnote_to_be_reexecuted":false,"source_hash":"cd59e126","execution_millis":2,"execution_start":1614115839355,"deepnote_cell_type":"code"},"source":"#!mkdir ~/.kaggle\n#!touch ~/.kaggle/kaggle.json\n\n#api_token = {\"username\":\"cmannnn\",\"key\":\"f4349f96544962460b7e7ec71ce8a3b5\"}\n\n#import json\n#with open('/root/.kaggle/kaggle.json', 'w') as file:\n#    json.dump(api_token, file)\n\n","execution_count":7,"outputs":[]},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00010-a6230db6-29ab-4c16-933d-a87c0c0704d5","deepnote_to_be_reexecuted":false,"source_hash":"7b7f5c1a","execution_millis":947,"execution_start":1614115839365,"deepnote_cell_type":"code"},"source":"!kaggle datasets download ywhenlyn/predict-future-sales-translated-to-english -p /predict_future_sales/files","execution_count":8,"outputs":[{"name":"stdout","text":"Traceback (most recent call last):\r\n  File \"/root/venv/bin/kaggle\", line 5, in <module>\r\n    from kaggle.cli import main\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/__init__.py\", line 23, in <module>\r\n    api.authenticate()\r\n  File \"/root/venv/lib/python3.7/site-packages/kaggle/api/kaggle_api_extended.py\", line 166, in authenticate\r\n    self.config_file, self.config_dir))\r\nOSError: Could not find kaggle.json. Make sure it's located in /root/.kaggle. Or use the environment method.\r\n","output_type":"stream"}]},{"cell_type":"code","metadata":{"tags":[],"cell_id":"00011-6d11df0c-1de3-4f53-bafc-326075ddd12d","deepnote_to_be_reexecuted":false,"source_hash":"a9a003e9","execution_millis":1279,"execution_start":1614115840318,"deepnote_cell_type":"code"},"source":"itemcats_df = pd.read_csv(\"/work/predict_future_sales/files/item_categories.csv\")\nitems_df = pd.read_csv(\"/work/predict_future_sales/files/items.csv\")\nsalestrain_df = pd.read_csv(\"/work/predict_future_sales/files/sales_train.csv\")\nshops_df = pd.read_csv(\"/work/predict_future_sales/files/shops.csv\")","execution_count":9,"outputs":[]},{"cell_type":"code","source":"","metadata":{"tags":[],"cell_id":"00010-149a7250-1a37-40f8-997c-4986d442e7fe","deepnote_cell_type":"code"},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"<a style='text-decoration:none;line-height:16px;display:flex;color:#5B5B62;padding:10px;justify-content:end;' href='https://deepnote.com?utm_source=created-in-deepnote-cell&projectId=44fa2d56-e957-4313-be50-5c4ecc704dff' target=\"_blank\">\n<img style='display:inline;max-height:16px;margin:0px;margin-right:7.5px;' src='data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iODBweCIgaGVpZ2h0PSI4MHB4IiB2aWV3Qm94PSIwIDAgODAgODAiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+CiAgICA8IS0tIEdlbmVyYXRvcjogU2tldGNoIDU0LjEgKDc2NDkwKSAtIGh0dHBzOi8vc2tldGNoYXBwLmNvbSAtLT4KICAgIDx0aXRsZT5Hcm91cCAzPC90aXRsZT4KICAgIDxkZXNjPkNyZWF0ZWQgd2l0aCBTa2V0Y2guPC9kZXNjPgogICAgPGcgaWQ9IkxhbmRpbmciIHN0cm9rZT0ibm9uZSIgc3Ryb2tlLXdpZHRoPSIxIiBmaWxsPSJub25lIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiPgogICAgICAgIDxnIGlkPSJBcnRib2FyZCIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTEyMzUuMDAwMDAwLCAtNzkuMDAwMDAwKSI+CiAgICAgICAgICAgIDxnIGlkPSJHcm91cC0zIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxMjM1LjAwMDAwMCwgNzkuMDAwMDAwKSI+CiAgICAgICAgICAgICAgICA8cG9seWdvbiBpZD0iUGF0aC0yMCIgZmlsbD0iIzAyNjVCNCIgcG9pbnRzPSIyLjM3NjIzNzYyIDgwIDM4LjA0NzY2NjcgODAgNTcuODIxNzgyMiA3My44MDU3NTkyIDU3LjgyMTc4MjIgMzIuNzU5MjczOSAzOS4xNDAyMjc4IDMxLjY4MzE2ODMiPjwvcG9seWdvbj4KICAgICAgICAgICAgICAgIDxwYXRoIGQ9Ik0zNS4wMDc3MTgsODAgQzQyLjkwNjIwMDcsNzYuNDU0OTM1OCA0Ny41NjQ5MTY3LDcxLjU0MjI2NzEgNDguOTgzODY2LDY1LjI2MTk5MzkgQzUxLjExMjI4OTksNTUuODQxNTg0MiA0MS42NzcxNzk1LDQ5LjIxMjIyODQgMjUuNjIzOTg0Niw0OS4yMTIyMjg0IEMyNS40ODQ5Mjg5LDQ5LjEyNjg0NDggMjkuODI2MTI5Niw0My4yODM4MjQ4IDM4LjY0NzU4NjksMzEuNjgzMTY4MyBMNzIuODcxMjg3MSwzMi41NTQ0MjUgTDY1LjI4MDk3Myw2Ny42NzYzNDIxIEw1MS4xMTIyODk5LDc3LjM3NjE0NCBMMzUuMDA3NzE4LDgwIFoiIGlkPSJQYXRoLTIyIiBmaWxsPSIjMDAyODY4Ij48L3BhdGg+CiAgICAgICAgICAgICAgICA8cGF0aCBkPSJNMCwzNy43MzA0NDA1IEwyNy4xMTQ1MzcsMC4yNTcxMTE0MzYgQzYyLjM3MTUxMjMsLTEuOTkwNzE3MDEgODAsMTAuNTAwMzkyNyA4MCwzNy43MzA0NDA1IEM4MCw2NC45NjA0ODgyIDY0Ljc3NjUwMzgsNzkuMDUwMzQxNCAzNC4zMjk1MTEzLDgwIEM0Ny4wNTUzNDg5LDc3LjU2NzA4MDggNTMuNDE4MjY3Nyw3MC4zMTM2MTAzIDUzLjQxODI2NzcsNTguMjM5NTg4NSBDNTMuNDE4MjY3Nyw0MC4xMjg1NTU3IDM2LjMwMzk1NDQsMzcuNzMwNDQwNSAyNS4yMjc0MTcsMzcuNzMwNDQwNSBDMTcuODQzMDU4NiwzNy43MzA0NDA1IDkuNDMzOTE5NjYsMzcuNzMwNDQwNSAwLDM3LjczMDQ0MDUgWiIgaWQ9IlBhdGgtMTkiIGZpbGw9IiMzNzkzRUYiPjwvcGF0aD4KICAgICAgICAgICAgPC9nPgogICAgICAgIDwvZz4KICAgIDwvZz4KPC9zdmc+' > </img>\nCreated in <span style='font-weight:600;margin-left:4px;'>Deepnote</span></a>","metadata":{"tags":[],"created_in_deepnote_cell":true,"deepnote_cell_type":"markdown"}}],"nbformat":4,"nbformat_minor":2,"metadata":{"orig_nbformat":2,"deepnote_notebook_id":"3a2e65a0-4d48-4e9b-8af7-9a7ffd822f4f","deepnote":{},"deepnote_execution_queue":[]}}